### PR TITLE
AutoRest++

### PR DESCRIPTION
**Pull request by ChiefMateStarbuck**
This PR adds Autorest functionality to the openAPI spec. 
It has been decided that the output of the code generation will live in their own repos, but for testing and discovery, it will be easier to work inside of the openAPI repo. 
This PR removes some redundancies that errored autorest. 
Autorest (e.g. `make autorest-python`) currently generates a python client with many naming redundancies. 
typescript does not work.

### CODE REVIEW

**Comment by ChiefMateStarbuck** [comment]
deploy!

**Comment by ChiefMateStarbuck** [comment]
!deploy

File: specification/resources/registry/responses/registry_options_response.yml [hunk 1]
```diff
@@ -14,23 +14,18 @@ content:
     schema:
       type: object
       properties:
-        options:
```
Comment (andrewsomething) [inline specification/resources/registry/responses/registry_options_response.yml:17]: Not sure I understand this change. The response body for `/v2/registry/options` has an `options` key:

```
curl -X GET -s   -H "Content-Type: application/json"   -H "Authorization: Bearer $DO_TOKEN"   "https://api.digitalocean.com/v2/registry/options" | jq .
{
  "options": {
    "subscription_tiers": [
      {
        "name": "Starter",
        "slug": "starter",
        "included_repositories": 1,
        "included_storage_bytes": 524288000,
        "allow_storage_overage": false,
        "included_bandwidth_bytes": 524288000,
        "monthly_price_in_cents": 0,
        "storage_overage_price_in_cents": 2,
        "eligible": false,
        "eligibility_reasons": [
          "OverRepositoryLimit"
        ]
      },
      {
        "name": "Basic",
        "slug": "basic",
        "included_repositories": 5,
        "included_storage_bytes": 5368709120,
        "allow_storage_overage": true,
        "included_bandwidth_bytes": 5368709120,
        "monthly_price_in_cents": 500,
        "storage_overage_price_in_cents": 2,
        "eligible": true
      },
      {
        "name": "Professional",
        "slug": "professional",
        "included_repositories": 0,
        "included_storage_bytes": 107374182400,
        "allow_storage_overage": true,
        "included_bandwidth_bytes": 107374182400,
        "monthly_price_in_cents": 2000,
        "storage_overage_price_in_cents": 2,
        "eligible": true
      }
    ],
    "available_regions": [
      "nyc3",
      "fra1",
      "ams3",
      "sfo3",
      "sgp1"
    ]
  }
}
```

Comment (scotchneat) [reply specification/resources/registry/responses/registry_options_response.yml:17]: +1

**Comment by ChiefMateStarbuck** [comment]
!deploy

**Comment by ChiefMateStarbuck** [comment]
!deploy

File: auto.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+input-file: tests/openapi-bundled.yaml 
+
+directive:
+  from: tests/openapi-bundled.yaml 
+  where: '$.paths["/v2/1-clicks"].get'
+  transform: >
+    $["operationId"] = "1Clicks_list";
```
Comment (danaelhe) [inline auto.yaml:7]: do we still want to keep this in even though we're not using directives to edit operationIDs or is this something else?

**Review by danaelhe — APPROVED** [review]
LGTM, just had one question

File: auto.yaml (hunk 2 above)
Comment (ChiefMateStarbuck) [reply auto.yaml]: removed

**Review by danaelhe — APPROVED** [review]
LGTM

**Pull request merged**